pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -308,6 +309,155 @@ func (am *Alertmanager) SyncAndApplyConfigFromDatabase() error {
 	return nil
 }
 
+func (am *Alertmanager) TestReceivers(ctx context.Context, c apimodels.TestReceiversConfig) (*apimodels.TestReceiversResult, error) {
+	// now represents the start time of the test
+	now := time.Now()
+	testAlert := &types.Alert{
+		Alert: model.Alert{
+			Labels: model.LabelSet{
+				model.LabelName("instance"): "foo",
+			},
+			Annotations: model.LabelSet{},
+			StartsAt:    now,
+		},
+		UpdatedAt: now,
+	}
+
+	// we must set a group key as this is required by some notification channels (i.e. webhook)
+	ctx = notify.WithGroupKey(ctx, testAlert.Labels.String())
+
+	tmpl, err := am.getTemplate()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get template: %w", err)
+	}
+
+	integrationsMap, err := am.buildIntegrationsMap(c.Receivers, tmpl)
+	if err != nil {
+		// The request contains invalid config for one or more contact points
+		return nil, fmt.Errorf("failed to create receivers: %w", err)
```

Comment:
Probably part of your plan per the comment above, but will mention it so that I don't forget: can we use a named error here to return this back as a 400?